torch/csrc/autograd/variable.cpp

```diff
@@ -368,4 +373,58 @@ unsigned VariableHooks::_register_hook(const Tensor& self, std::function<Tensor(
   return idx;
 }
 
+TORCH_API void handle_view_on_rebase(DifferentiableViewMeta* diff_view_meta, bool indirect) {
+  /// See NOTE [ View + Inplace detection ] for justification of the logic below
+  if (diff_view_meta->allow_rebase_history != OnRebase::ALLOW_REBASE) {
+    auto grad_fn = diff_view_meta->grad_fn_.get();
+    std::string msg;
+    std::string modified_obj;
+    // Create the header for the error message.
+    if (indirect) {
+      modified_obj = "its base or another view of its base has been";
+    } else {
+      modified_obj = "is being";
+    }
+    if (grad_fn) {
+      msg = c10::str("Output ", diff_view_meta->output_nr_, " of ", grad_fn->name(), " is a view and ",
+                     modified_obj, " modified inplace.");
+    } else {
+      msg = c10::str("A view was created in no_grad mode and ", modified_obj, " modified inplace with grad mode enabled.");
+    }
+
+    if (diff_view_meta->allow_rebase_history == OnRebase::ERROR_REBASE) {
+      TORCH_CHECK(false, msg, " This view is the output of a function that returns multiple views. Such functions do not"
+                         " allow the output views to be modified inplace.");
+    } else {
+      if (diff_view_meta->allow_rebase_history == OnRebase::WARN_REBASE) {
+        TORCH_INTERNAL_ASSERT(!grad_fn);
+        msg = c10::str(msg, " Given that this use case is ambiguous and error-prone, it is deprecated and will be forbidden"
+                       "  starting 1.6. You can clarify your code and remove this warning by moving both the view and the"
+                       " inplace either both inside the no_grad block (if you don't want the inplace to be tracked) or both"
```

Comment:
I guess one reason you wrote the table, is so that you could convince yourself that you could assume that a `no_grad` block was involved, when you end up with a `OnRebase::WARN_REBASE` setting on the enum. But this is a horribly indirect way of representing the state, when it could be represented in a much more simple way (once again, getting to the idea of thinking about the "semantics"): just straight up **record** whether or not a view was allocated in the no grad. None of this "inferring it based on the intent of the rebase warning: oh, *if* I see a `WARN_REBASE`, that must have come from a `no_grad`." What if someone else adds another site where `WARN_REBASE` is introduced? Now you have to fix this error message.

Reply:
Right now, you are storing an 'operational' piece of data, whether or not to warn on rebase for a view. Instead, store a 'semantic' piece of data, e.g., where the view came from, and then interpret the semantics when it matters (when you rebase.)

Reply:
That sounds good. I will change it to `CreationMeta` and save flags like `DEFAULT, IN_CUSTOM_FUNCTION, MULTI_OUTPUT_CODEGEN, NO_GRAD_MODE` to make it clearer.

Reply:
Sounds good to me.